### PR TITLE
Update the protocol to version 3.17

### PR DESCRIPTION
**Pull request by CGNonofr**
Notes:
- InlineValue feature is not yet supported by monaco, and it's not possible to NOT register the feature

### CODE REVIEW

**Comment by CGNonofr** [comment]
This code is not tested yet

And I'm not sure about 440140aa03e0541fab580fde54eb822526d3188f and why it was required

**Comment by kaisalmen** [comment]
@CGNonofr Thank you. I will have a look later today.

**Comment by CGNonofr** [comment]
> @CGNonofr Thank you. I will have a look later today.

Ok NOW it's tested and ready to be reviewed

File: packages/client/src/monaco-language-client.ts
Comment (kaisalmen) [inline packages/client/src/monaco-language-client.ts:91]: Does the base implementation now perform this registration and it is therefore no longer needed here?

Comment (kaisalmen) [reply packages/client/src/monaco-language-client.ts:91]: Ok, I looked at `vscode-languageclient\lib\common\client.js` and this is done there. Was that an error to register it twicebefore or was that really an change in the lc lib?

**Review by kaisalmen — APPROVED** [review]
Comments don't imply required changes, but please look at the null check. If that needs ok, LGTM

**Comment by kaisalmen** [comment]
@CGNonofr sorry it took a bit longer. The week didn't turn out as planned. 😅

File: packages/client/src/monaco-language-client.ts
Comment (CGNonofr) [reply packages/client/src/monaco-language-client.ts:91]: I don't really know, it seems there isn't any consequence of registering a feature twice, so I don't know for how long it was there.

I think these features were `proposed` back then, so we added them and forgot to remove it when they became builtin

File: packages/client/src/monaco-language-client.ts
Comment (CGNonofr) [reply packages/client/src/monaco-language-client.ts:18]: I'm not sure to understand what you mean, can you provide the code you want to be there?

Comment (kaisalmen) [reply packages/client/src/monaco-language-client.ts:18]: Shouldn't `token != null` be`token !== null`?

Comment (CGNonofr) [reply packages/client/src/monaco-language-client.ts:18]: Do you want me to change it?

Comment (kaisalmen) [reply packages/client/src/monaco-language-client.ts:18]: Ok, so it can only just be:
```js
token ? Promise.resolve(result || undefined) : (result || undefined);
```
because the Typescript says:
```js
token?: vscode.CancellationToken | undefined
```
So, it exists or doesn't.
But, of course with JavaScript someone can throw in `null`, so if it `null` or `undefined` should be treated the same, then just leave it code as it is and sorry for the confusion. If you think types are the law, then make it strict. :-)

Comment (CGNonofr) [reply packages/client/src/monaco-language-client.ts:18]: I'm used to strict eslint rules, so doing a if on something else than a boolean make me crazy now :scream:

Comment (kaisalmen) [reply packages/client/src/monaco-language-client.ts:18]: This is a boolean expression, if is defined, then it is true: `token ? Promise.resolve(result || undefined) : (result || undefined);`

Comment (kaisalmen) [reply packages/client/src/monaco-language-client.ts:18]: > Can we release this one or do you want to wait for the next version?

I am just now working on bringing the webpack config for the web client example back. If both are in, then we can release a new version. Should we consider investigating the new issue that came up before. WDYT?

Comment (kaisalmen) [reply packages/client/src/monaco-language-client.ts:18]: I mean a real new version `0.19.0` or shall we call it `1.0.0`?

Comment (CGNonofr) [reply packages/client/src/monaco-language-client.ts:18]: > This is a boolean expression, if token is defined, then it is true: token ? Promise.resolve(result || undefined) : (result || undefined);

I'm not sure to understand what you mean, `token` is a boolean?

> Should we consider investigating the new issue that came up before. WDYT

Are you talking about https://github.com/TypeFox/monaco-languageclient/issues/351? I don't think it's a new issue, but I'm currently investigating. There is probably no reason for it to block the next release tho

> I mean a real new version 0.19.0 or shall we call it 1.0.0?

I never understand why so many libraries are using a version number < 1 even after years (even monaco!).
But it's not a big deal, as you wish!

Comment (kaisalmen) [reply packages/client/src/monaco-language-client.ts:18]: Let's aim for 1.0.0. webpack example is almost working again. `monaco` workers are not working properly. Will fix this today. Release today or tomorrow.

Comment (CGNonofr) [reply packages/client/src/monaco-language-client.ts:18]: > You can rewrite this as such (token is undefined => false or defined => true)
That's what I meant when talking about `if on anything else than boolean`, relying on the trythiness of an object and the falsiness of undefined seems wrong and dangerous, no big deal tho since the whole project is done this way

Comment (kaisalmen) [reply packages/client/src/monaco-language-client.ts:18]: Please press the merge button if you like. 👍

**Pull request merged**
